### PR TITLE
fix: errorCodes import using ESM

### PR DESCRIPTION
**Pull request by melroy89**
- [x] Fixing errorCodes import using ESM. 
- [x] Adding an unit-test
- [x] Extend `.taprc` configuration to run `test.mjs` files as well
- [ ] Rename other mjs files to `.test.mjs` filename?

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

In favor of: https://github.com/fastify/fastify/pull/5388

### CODE REVIEW

**Review by jsumners — CHANGES_REQUESTED** [review]
What is this fixing? Please provide context when opening pull requests. Also, this is missing a test.

**Comment by melroy89** [comment]
It's indeed missing a test.

**Comment by climba03003** [comment]
> What is this fixing? Please provide context when opening pull requests. Also, this is missing a test.

It is actually fixing how people can be import `errorCodes` in ESM.

```js
// Before
import Fastify from 'fastify'
const { errorCodes } = Fastify

// After
import { errorCodes } from 'fastify'
```

The reason is behind how Node.js transform CJS to ESM named export.
Not going to explain the detail, but it requires the syntax of `module.export.<named export> = <value>`

File: fastify.js [hunk 1]
```diff
@@ -893,8 +893,6 @@ function fastify (options) {
   }
 }
 
-fastify.errorCodes = errorCodes
```
Comment (jsumners) [inline fastify.js:896]: This is a breaking change.

Comment (climba03003) [reply fastify.js:896]: Actually, Not.

```js
fastify.errorCodes = errorCodes
module.exports = fastify
// is the same as 
module.exports = fastify
module.exports.errorCodes = errorCodes
```

It is all modifying the same `fastify` object.

Comment (melroy89) [reply fastify.js:896]: And currently importing the errorCodes in the current implementation is broken when using ESM.

**Review by Uzlopak — APPROVED** [review]
LGTM

**Comment by melroy89** [comment]
> It is actually fixing how people can be import `errorCodes` in ESM.

Important to mention this "after" is already documented on your fastify website (it's not yet working like this): https://fastify.dev/docs/latest/Reference/Errors/#fastify-error-codes

This pr try to solve that, so it's inline with the documentation.

**Comment by Uzlopak** [comment]
What we need is a test. So basically create a esm-error-codes.test.mjs and import the errorCodes as mentioned in the documentation and check if it is not undefined.

**Comment by melroy89** [comment]
> What we need is a test. So basically create a esm-error-codes.test.mjs and import the errorCodes as mentioned in the documentation and check if it is not undefined.

Agreed. Will do. Give me one second.

**Comment by melroy89** [comment]
There are TS Type tests in `tests/types` folder (incl. errors.tst-d.ts), how to execute those tests? Since `tap` doesn't seem to run them when executing `npm run test`

**Comment by Uzlopak** [comment]
The typescript typings tests are Independent from this issue. Just because you test it with tsd, you wont test it in node.

**Comment by melroy89** [comment]
It seems that the test in `esm` directory are not all executed (only `test/esm/index.test.js `). Which is a problem!

I believe tap is only executing `.js` files.

**Comment by melroy89** [comment]
Should I extend the `.taprc` file?

**Comment by melroy89** [comment]
Should I rename the other tests in the `esm` directory to comply with the same filename convention for tests (aka `.test.mjs` in this case).? So those tests are getting executed by `tap` as well? Since now only my `errorCodes.test.mjs` test is getting executed. The other tests were never executed before via tap.. Basically dead tests.

**Review by climba03003 — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]
LGTM

**Comment by melroy89** [comment]
Thanks for merging. I still would like to know what you want to do with those other ESM tests that were never part of tap?

**Comment by Uzlopak** [comment]
It can not harm to rename them to mjs.

**Comment by melroy89** [comment]
> It can not harm to rename them to mjs.

You mean `.test.mjs` as suffix (they already have mjs as "extension"), but I use the same file syntax as for your other tests: `.test.js` and `.test.mjs`. Currently those ESM files aren't executed (they were never executed actually), only my `.test.mjs` test :).

But OK, I think I will create a follow-up PR to fix those other ESM tests.

**Pull request merged**